packages/angular/cli/src/commands/mcp/mcp-server.ts

```diff
@@ -46,18 +50,43 @@ export async function createMcpServer(context: {
   server.registerTool(
     'list_projects',
     {
-      title: 'List projects',
+      title: 'List Angular Projects',
       description:
-        'List projects within an Angular workspace.' +
-        ' This information is read from the `angular.json` file at the root path of the Angular workspace',
+        'Lists the names of all applications and libraries defined within an Angular workspace. ' +
+        'It reads the `angular.json` configuration file to identify the projects. ' +
+        'By default, it searches from the current working directory, but a different path can be specified.',
+      inputSchema: {
+        workingDirectory: z
+          .string()
+          .optional()
+          .describe(
+            'Optionally use a filesystem directory as the initial location to search for an Angular Workspace root path.' +
+              ' The root path is determined by the presence of an `angular.json` file.',
+          ),
+      },
+      annotations: {
+        readOnlyHint: true,
+      },
     },
-    () => {
-      if (!context.workspace) {
+    async ({ workingDirectory }) => {
+      let workspace: AngularWorkspace | undefined;
+      if (!workingDirectory || workingDirectory === context.workspace?.basePath) {
+        workspace = context.workspace;
+      } else {
+        try {
+          workspace = await AngularWorkspace.load(path.join(workingDirectory, 'angular.json'));
```

Comment:
I am a bit confused by this branch and why is it needed, `context.workspace` has the contents of the `angular.json` and is undefined when the workspace does not exist.